Jenkinsfile

```diff
@@ -0,0 +1,116 @@
+pipeline {
+    agent {
+        ecs {
+            inheritFrom 'terraform'
+        }
+    }
+    environment {
+        stage = getStageFromBranch()
+        account_number = getAccountNumberFromBranch()
+    }
+    stages {
+        stage('Configure AWS credentials') {
+            steps {
+                echo 'Configuring AWS credentials...'
+                sh 'java -version'
+                sh "aws configure set role_arn arn:aws:iam::${env.account_number}:role/${env.stage}-terraform-role --profile ${env.stage}terraform"
+                sh "aws configure set region eu-west-2 --profile ${env.stage}terraform"
+                sh "aws configure set source_profile default --profile ${env.stage}terraform"
+                withCredentials([[
+                                         $class: 'AmazonWebServicesCredentialsBinding',
+                                         credentialsId: "${env.stage}Terraform",
+                                         accessKeyVariable: 'AWS_ACCESS_KEY_ID',
+                                         secretKeyVariable: 'AWS_SECRET_ACCESS_KEY'
+                                 ]]) {
+                    sh 'aws configure set aws_access_key_id ${AWS_ACCESS_KEY_ID}'
+                    sh 'aws configure set aws_secret_access_key ${AWS_SECRET_ACCESS_KEY}'
+                    sh 'cat ~/.aws/config'
+                    sh 'cat ~/.aws/credentials'
+                }
+                sh 'export AWS_PROFILE=default'
+            }
+        }
+        stage('Set up Terraform workspace') {
+            //no-color option set for Terraform commands as Jenkins console unable to output the colour
+            //making output difficult to read
+            steps {
+                echo 'Initializing Terraform...'
+                sh 'terraform init -no-color'
+                //If Terraform workspace exists continue
+                sh "terraform workspace new ${env.stage} -no-color || true"
```

Comment:
It looks like you can simplify all the commands by removing `-no-color` if you set a `TF_CLI_ARGS` environment variable at the start of the pipeline: https://www.terraform.io/docs/commands/environment-variables.html#tf_cli_args-and-tf_cli_args_name